test/integration.js

```diff
@@ -0,0 +1,182 @@
+                const driver = new webdriver.Builder().forBrowser(browser).build();
+                driver.get(url).then(function() {
+                    return driver.findElement(webdriver.By.id(newsletterSubmitId));
+                }).then(function(button) {
+                    return button.submit();
+                }).then(function() {
+                    return driver.findElement(webdriver.By.css(`#${emailInputId}:invalid`));
+                }).then(function(invalid) {
+                    done();
+                }, function(valid) {
+                    done(new Error('Form not marked as invalid with missing email'));
+                });
+            });
+
+            test.it('should fail if policy checkbox isn\'t checked', function(done) {
+                const driver = new webdriver.Builder().forBrowser(browser).build();
+                driver.get(url).then(function() {
+                    return driver.findElement(webdriver.By.id(newsletterSubmitId));
+                }).then(function(button) {
+                    return button.submit()
+                }).then(function() {
+                    driver.findElement(webdriver.By.css(`#${policyCheckboxId}:invalid`));
+                }).then(function(invalid) {
+                    done();
+                }, function(valid) {
+                    done(new Error('Form not marked as invalid with missing policy checkbox'));
+                });
+            });
+
+            test.it('should display a "Thanks" message on successful submission', function(done) {
+                let expectedText = 'Thanks! Please check your inbox to confirm your subscription.';
+
+                const driver = new webdriver.Builder().forBrowser(browser).build();
+
+                driver.get(url).then(function() {
+                    return driver.findElement(webdriver.By.id(emailInputId));
+                }).then(function(email) {
+                    return email.sendKeys('vs_integration_tester@mailinator.com');
+                }).then(function() {
+                    return driver.findElement(webdriver.By.id(policyCheckboxId));
+                }).then(function(policy) {
+                    return policy.click();
+                }).then(function() {
+                    return driver.findElement(webdriver.By.id(newsletterSubmitId));
+                }).then(function(button) {
+                    return button.submit();
+                }).then(function() {
+                    driver.wait(
+                        function() {
+                            return driver.findElement(webdriver.By.id('newsletter')).then(function(body) {
+                                return body.getText();
+                            }).then(function(text) {
+                                text = text.toLowerCase();
+                                expectedText = expectedText.toLowerCase();
+                                return text.indexOf(expectedText) > -1;
```

Comment:
I changed the logic here to check if the div is visible. I was holding out hope that I could make this work across all circumstances (one of which is IE, which posts to the generic newsletter form at mozilla.org/about/newsletter, and therefore gets different markup but the same "Thanks" string). But right now these tests aren't designed to run on windows. If they ever are we can implement a switch there.